tests/unit/v2/cache/testMemoryBackend.js

```diff
@@ -0,0 +1,118 @@
+
+describe('Test cache memory backend', () => {
+    let mem;
+    beforeEach(() => {
+        mem = new MemoryCache();
+    });
+
+    it('should create a new memory backend instance', () => {
+        assert.equal(mem instanceof MemoryCache, true);
+    });
+
+    testValues.map(event =>
+        it('should set a key', () => {
+            const key = schema.getUtapiMetricKey('test', event);
+            return mem.setKey(key, event).then(res => {
+                assert.strictEqual(true, res);
+                const db = mem._data;
+                const expectedRes = {};
+                expectedRes[key] = event;
+                assert.deepStrictEqual(db, expectedRes);
+            });
+        }));
+
+    testValues.map(event =>
+        it("should get a key's value", () => {
+            const key = schema.getUtapiMetricKey('test', event);
+            return mem
+                .setKey(key, event)
+                .then(() => mem.getKey(key))
+                .then(res => {
+                    assert.strictEqual(res, event);
+                });
+        }));
+
+    testValues.map(event =>
+        it('should add a key to a shard', () => {
+            const shard = shardFromTimestamp(event.timestamp);
+            return mem.addToShard(shard, event).then(res => {
+                assert.strictEqual(true, res);
+                const db = mem._shards;
+                const expectedRes = {};
+                expectedRes[shard] = [schema.getUtapiMetricKey('utapi', event)];
+                assert.deepStrictEqual(db, expectedRes);
+            });
+        }));
+
+    it('should fetch the keys in a shard', () => {
+        const shard = shardFromTimestamp(new Date().getTime());
+        return Promise.all(testValues.map(event => mem.addToShard(shard, event)))
+            .then(() => mem.getKeysInShard(shard))
+            .then(keys => {
+                const db = mem._shards;
+                assert.deepStrictEqual(keys, db[shard]);
+                assert.deepStrictEqual(
+                    keys,
+                    testValues.map(e => schema.getUtapiMetricKey('utapi', e)),
+                );
+            });
+    });
+
+    it('should delete a shard and its keys', () => {
+        const shard = shardFromTimestamp(new Date().getTime());
+        return Promise.all(testValues.map(event => mem.addToShard(shard, event)))
+            .then(() => mem.deleteShardAndKeys(shard))
+            .then(res => {
+                assert.strictEqual(res, true);
+                const shards = mem._shards;
+                const db = mem._data;
+                assert.deepStrictEqual(undefined, shards[shard]);
+                assert.deepStrictEqual(db, {});
+            });
+    });
+
+    it('should get all shards', () => Promise.all(
+        testValues.map(event => {
+            const shard = shardFromTimestamp(event.timestamp);
+            return mem.addToShard(shard, event.uuid);
+        }),
+    )
+        .then(() => mem.getShards())
```

Comment:
The end parenthesis for promise.all makes it look like the it test is closed way before it actually is. Had me confused for a good few minutes.